src/rebar_app_discover.erl

```diff
@@ -48,7 +48,9 @@ merge_deps(AppInfo, State) ->
     State1 = lists:foldl(fun(Profile, StateAcc) ->
                                  AppProfDeps = rebar_state:get(AppState, {deps, Profile}, []),
                                  TopLevelProfDeps = rebar_state:get(StateAcc, {deps, Profile}, []),
-                                 ProfDeps2 = lists:keymerge(1, TopLevelProfDeps, AppProfDeps),
+                                 ProfDeps2 = dedup(lists:keymerge(1,
```

Comment:
Or wait, the dedup matches on the entire tuple? So this whole thing would be the same as a `umerge`?